hashview/models.py

```diff
@@ -21,24 +24,78 @@ class Users(db.Model, UserMixin):
     admin = db.Column(db.Boolean, nullable=False, default=False)
     pushover_app_id = db.Column(db.String(50), nullable=True)
     pushover_user_key = db.Column(db.String(50), nullable=True)
+    last_login_utc = db.Column(db.DateTime, nullable=True, default=datetime.utcnow)
     wordlists = db.relationship('Wordlists', backref='tbd', lazy=True)
     rules = db.relationship('Rules', backref='owner', lazy=True)
     jobs = db.relationship('Jobs', backref='owner', lazy=True)
     tasks = db.relationship('Tasks', backref='owner', lazy=True)
     taskgroups = db.relationship('TaskGroups', backref='owner', lazy=True)
 
-    def get_reset_token(self, expires_sec=1800):
-        s = Serializer(Config.SECRET_KEY, expires_sec)
-        return s.dumps({'user_id': self.id}).decode('utf-8')
+    def _get_reset_token_salt(self) -> str:
+        """
+        Create salt data for password reset token signing. The return value will be hashed
+        together with the signing key. This ensures that changes to any of the fields included
+        in the salt invalidates any tokens produced with the old values.
+        """
+        return json.dumps([
+            self.first_name,
+            self.last_name,
+            self.password.hash.decode() if self.password is not None else '',
+            self.last_login_utc.to('UTC').isoformat() if self.last_login_utc else None
```

Comment:
This throws the following error:
```
AttributeError: 'datetime.datetime' object has no attribute 'to'
```

could we just do `self.last_login_utc if self.last_login_utc else None` ? do we need to convert the format if its all just going to be mashed into a salt?

Reply:
Ah, i see now why it needs to be converted :)

If i try what i suggested above i get the following error:
`TypeError: Object of type datetime is not JSON serializable`

so the datetime does need to be converted

Reply:
We should be able to drop the "to" since it is stored as UTC, and just covert to isoformat to then be serialized as a string.